File: app/src/main/java/org/salient/artvideoplayer/DensityUtil.java
package org.salient.artvideoplayer;

import android.app.Activity;
import android.content.Context;
import android.util.DisplayMetrics;
import android.util.Log;

/**
 *  Created by Mai on 2018/5/18
 * *
 *  Description:屏幕像素转换类
 * *
 */
public class DensityUtil {

    /**
     * dip转像素
     */
    public static int dip2px(Context context, int dip) {
        final float SCALE = context.getResources().getDisplayMetrics().density;
        return (int) ((float) dip * SCALE + 0.5f);
    }

    /**
     * 像素转dip
     */
    public static float px2dip(Context context, int Pixels) {
        final float SCALE = context.getResources().getDisplayMetrics().density;
        return Pixels / SCALE;
    }

    /**
     * 屏幕分辨率宽
     */
    public static int getWindowWidth(Activity activity) {
        DisplayMetrics dm = new DisplayMetrics();
        activity.getWindowManager().getDefaultDisplay().getMetrics(dm);
        return dm.widthPixels;
    }

    /**
     * 屏幕分辩类高
     */
    public static int getWindowHeight(Activity activity) {
        DisplayMetrics dm = new DisplayMetrics();
        activity.getWindowManager().getDefaultDisplay().getMetrics(dm);
        return dm.heightPixels;
    }

    /**
     * 屏幕的dpi
     */
    public static float getDmDensityDpi(Activity activity) {
        DisplayMetrics dm = new DisplayMetrics();
        activity.getWindowManager().getDefaultDisplay().getMetrics(dm);
        return dm.density;
    }
}

File: app/src/main/java/org/salient/artvideoplayer/DensityUtil.kt
package org.salient.artvideoplayer

import android.app.Activity
import android.content.Context
import android.util.DisplayMetrics

/**
 * Created by Mai on 2018/5/18
 * *
 * Description:屏幕像素转换类
 * *
 */
object DensityUtil {
    /**
     * dip转像素
     */
    fun dip2px(context: Context, dip: Int): Int {
        val SCALE = context.resources.displayMetrics.density
        return (dip.toFloat() * SCALE + 0.5f).toInt()
    }

    /**
     * 像素转dip
     */
    fun px2dip(context: Context, Pixels: Int): Float {
        val SCALE = context.resources.displayMetrics.density
        return Pixels / SCALE
    }

    /**
     * 屏幕分辨率宽
     */
    fun getWindowWidth(activity: Activity): Int {
        val dm = DisplayMetrics()
        activity.windowManager.defaultDisplay.getMetrics(dm)
        return dm.widthPixels
    }

    /**
     * 屏幕分辩类高
     */
    fun getWindowHeight(activity: Activity): Int {
        val dm = DisplayMetrics()
        activity.windowManager.defaultDisplay.getMetrics(dm)
        return dm.heightPixels
    }

    /**
     * 屏幕的dpi
     */
    fun getDmDensityDpi(activity: Activity): Float {
        val dm = DisplayMetrics()
        activity.windowManager.defaultDisplay.getMetrics(dm)
        return dm.density
    }
}